pptxApp.py

```diff
@@ -0,0 +1,158 @@
+# -----------------------------MODULE SECTION-------------------------------------------------------------------
+import time
+from pptx import Presentation
+import openai
+import json
+import os
+import re
+import asyncio
+
+# -----------------------------CONSTANT SECTION-------------------------------------------------------------------
+openai.api_key = os.environ.get('API_KEY')
+CONTENT = [
+    {"role": "system", "content": "Can you explain the slides in basic english, and provide examples if needed!"}
+]
+ERROR_MESSAGE = "Something is wrong:"
+ENGINE_MODEL = "gpt-3.5-turbo"
+
+
+# -----------------------------METHOD SECTION-------------------------------------------------------------------
+async def parse_presentation(presentation_path):
+    """
+    this method receives a path for a pptx presentation, checks if the path is found in the operating system, if so,
+    parses the data to slides. the method, returns a list of explanations.
+    :param: presentation_path: path of a power-point presentation (String)
+    :return: list of explanations (List of strings)
+    """
+    if not os.path.isfile(presentation_path):  # check if path is available
+        print(f"{ERROR_MESSAGE} the path you provided does not exist.")
+        return []
+
+    prs = Presentation(presentation_path)
+    explanations = []
+    for slide in prs.slides:
+        explanation = await parse_slide_of_pptx(slide)  # call method to parse data in each slide
+        explanations.append(explanation)
+    return explanations
+
+
+# --------------------------------------------------------------------------------------------------------------------
+async def parse_slide_of_pptx(slide):
+    """
+    this method receives a single slide, the method parses the data on that slide, it calls another method to get the
+    response and return it to the parse_presentation method. it throws an error if an exception occurred.
+    :param: slide: a single slide from the power-point (slide object)
+    :return: the explanation if the processing went well, an error, otherwise. (List of Strings)
+    """
+    try:
+        response = await request_completion(slide)  # call a function to get explanations on each slide.
+        return response
+    except Exception as error:  # if an error occurred, display a suitable message to user.
+        error_message = f"{ERROR_MESSAGE} Error processing slide: {str(error)}"
+        return error_message
+
+
+# --------------------------------------------------------------------------------------------------------------------
+def parse_text_of_slide(slide):
+    """
+    this method receives a slide, and extracts all the extractable text found in that slide. In addition, it cleans
+    the data using the strip function.
+    :param: slide: a single slide from the power-point (Slide Object)
+    :return: explanation response (String)
+    """
+    slide_text = []
+    for shape in slide.shapes:
```

Comment:
there's too much for-loops, it's bad complexity.
find a better way to implement your logic.